src/middleware/normalize.rs

```diff
@@ -50,13 +72,18 @@ where
         ok(NormalizePathNormalization {
             service,
             merge_slash: Regex::new("//+").unwrap(),
+            trim_last_slash: match self.0 {
+                TrailingSlash::Trim => true,
+                TrailingSlash::Always => false,
+            },
         })
     }
 }
 
 pub struct NormalizePathNormalization<S> {
     service: S,
     merge_slash: Regex,
+    trim_last_slash: bool,
```

Comment:
Given how we're using it, makes sense to keep the enum here too. Makes for more expressive matches later and possible ease of extensibility.